storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/put/PutCollectionResult.java

```diff
@@ -0,0 +1,105 @@
+package com.pushtorefresh.storio.contentprovider.operation.put;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Representation of result of Put Operation for collection of objects
+ * <p/>
+ * Instances of this class are Immutable
+ *
+ * @param <T> type of objects
+ */
+public class PutCollectionResult<T> {
+
+    @NonNull
+    private final Map<T, PutResult> results;
+
+    @Nullable
+    private volatile Integer numberOfInsertsCache;
+
+    @Nullable
+    private volatile Integer numberOfUpdatesCache;
+
+    private PutCollectionResult(@NonNull Map<T, PutResult> putResults) {
+        this.results = Collections.unmodifiableMap(putResults);
+    }
+
+    /**
+     * Creates new instance of {@link PutCollectionResult}
+     *
+     * @param putResults results of Put Operation
+     * @param <T>        type of objects
+     * @return immutable instance of {@link PutCollectionResult}
+     */
+    @NonNull
+    public static <T> PutCollectionResult<T> newInstance(@NonNull Map<T, PutResult> putResults) {
+        return new PutCollectionResult<>(putResults);
+    }
+
+    /**
+     * Returns immutable Map of pairs (object, PutResult)
+     *
+     * @return immutable Map of pairs (object, PutResult)
+     */
+    @NonNull
+    public Map<T, PutResult> results() {
+        return results;
+    }
+
+    /**
+     * Returns number of inserts from all {@link #results()}
+     *
+     * @return number of inserts from all {@link #results()}
+     */
+    public int numberOfInserts() {
+        final Integer cachedValue = numberOfInsertsCache;
+
+        if (cachedValue != null) {
+            return cachedValue;
+        }
+
+        int numberOfInserts = 0;
+
+        for (T object : results.keySet()) {
+            if (results.get(object).wasInserted()) {
```

Comment:
Hm I mean, that 
`for (PutResult result : results.values()) {` 
will be faster then current variant
`for (T object : results.keySet()) {
PutResult result = results.get(object)`